.travis.yml

```diff
@@ -0,0 +1,65 @@
+    - secure: "J61/KUCNGuWiZ6kiZDMjcKWFXGfnluE2+umVV8zgKX0MmGMv5ZyB1QIBNgQi72UUZSrvCUFh+lf77t7SgdHVM2hrsAITgFraFIl3Cyd3nfSoyoI2g7clhtar3rOovyg+hmScVDy96xeIgbT+DOzdEqKw5AEanQO9wZxI+X6jUcqOJsj8dTVPJwaNOfQ/416qQxGQXKQMyBRn9N6e/Im0UmpfifqYLvnWK5+ELhxuFNs7ZZd/nzQRgA97u9ytDzHC1SIIK2LPLgn64IaiBdau8jWqD2PiveYlKzLEAu7aiDkLF7KUcOkh8Q3UUbf0sJt3IO0TJRUe0hdvcbpHpgHHQlTdKUuTVijpN9spdWobuLIp1QUug+zv1vlFUvypt1wR80wmiv9xFPYHrOvV7hTksNReo6yDeIE2H/YSLQSSKJaWfvT9vXDxMj4aNb3UVWp0FTdmd4d0xFG94pxYImXI/ZyzFqPES33MDDSC8ffIJfyX9MmNsEl3e6PaeWrgulp4Nkhene25R+kLrosqw+cTPoc92Hctm6zndrbUVeiywBPbRjXiEWakJ2MDzAqzd4GP/1tuxIL1YY8MOpKw4SNV19Ke1ywy59iRAb8GY4nj66aQOgErOdkGla8s69KJrr221rh4PusisYDRrirMev4MS0b68snxNjcN0FEk+IBvlL4="
+  # Heroku
+    - herokuapp: rajrahmantodoapp
+    #apikey
+    - secure: "DKYA4sYt6Qet39+erw7L/RMSgBp3Up0KUk51SUnddzZAizg3wWy+vrWyjM6PN/B4AtlKpYiJORf0ENvMtLO6q0IiT0aM3u+Ha7CDgcG10F8cBYvQSVhUh5BXoX4+bsjn3D7e+ldTAfMOzqrHDUIVsi1xe2Mog6HNSlHu+MW6Q+UtFAQ8Q7qBS2zknVXLhtkzoMw64czqvAuM8TXFi+fSOeeWlbZx9fYlZmieLZ2giOClb8+3Y+e/zGS2EmWtD1Welb7byjJUh4Vq7k0lvWDQy80z1K1AHBuJI4BHvoJ8Zlup51P9OhfOblbpDCvU+L78rYsHLnPwLablak0nxdcuxbtKsv/kMl22hXt2ZLNb4C9x5WYbhCgPy1Ur2o9D+Tj/4VT5qo4HO0UPdPeL/2m5j8sFjk3MB2zYx6UEGrv79CKZ7EZJwGdcswijGGva+/e5CIVk7svZM6gQJfmoDrHmCFSbkPkcDX/cgtgakOS1CH5Kx5O6e6vOx2vGAcMAqxpv4LWP6pVzAbhmuwE46FyvA8M+CfzYsnWYRz4lqnVQxjf2RRIIhZ/idrSHc0Qgs1MWrvsZX2YPUc2jU38pInQMVh+2VFjuRjBW2f6T/so+CgcSEeW62Sl3epwsuHYodbI7Nqb9X5auwbS80xlO8XUu4LECY7x9NUrOsaUTMSU7hPU="
+    #heroku_login
+    - secure: "J6O8nLCQq8Qr4fkXRtd9kc0kp5UNymXcWZk6d2j/SGHWLiNxsgkGNVOMv8L5SqYCx5GfmUC7Tvag72f8+TIc2BiznlCAgia5kXY2syWxhwxhT5bvXXGoRL9j53W20FvjHWwUMiwTnqV4d+1eSpqWGoWCtd/kT5XiGgXQW3I2d6R4OzhOjNR2UcZEYhu3U6o5Zn5pZPzfFgaqMdk6iKoa5KFwBR+1wLJYOndlp0SlsNBjbtnDfy2GrqeZiEIjnn3HjyhPkAqWBtEGkut167JVYQRbJ1mC+puStpZeK0oRBh4Um5i9ufYCKnOqi/lRmH6oD8hHPXRY2P2ZlY4c1FcfflseoT36/Zlh78/LZBKVSFd0Z/LYPr3eWylHF+0gfbxA4zda+cgcVlPiOgt52OZopk3RQm19CHnMQSAuxhCpCii+zDB9dYJ+dA5+TtWCF1XEx1G4efNFDJIFw6A/ix5Pu+/vHdkA0XJnQd5WXPuHJmdKKN5NaRA2TiHeS2rwJhWLm+zKCfTBhCWyxMHHUOmq2P/IejeAXab1Fcq7t4oOmX86UXUkAc6lyCmH8QcG36uYYKqZbRDk4lvy0HEOnp1MdX1yuAr4SrUjfYixrAq846DASyszgp2Ou6m8WtpcVGx+qz6Hh/ByfrTM2YCkOSBndj7Jkc7d933AfYjf/4puIVI="
+
+services:
+  - docker
+
+before_script:
+  - curl https://cli-assets.heroku.com/install.sh | sh
+  - docker login --username $docker_login --password $docker_password
+
+script:
+
+  # Docker TEST Build the Image
+  - docker build --target test --tag todoapp:test .
+
+  # Docker Run Unit Test
+  - echo "Running Unit Tests"
+  - docker run todoapp:test test_view_model.py
+
+  # Docker Run Integration tests 
+  - echo "Running Integration Tests"
+  #- docker run todoapp:test test_todoapp.py
+
+  # Docker Run E2E 
+  - echo "Running E2E Tests"
+  - docker run -e MONGO_DB_NAME=$MONGO_DB_NAME -e MONGO_CONN=$MONGO_CONN -e MONGO_LIST_TODO=$MONGO_LIST_TODO -e MONGO_LIST_INPROGRESS=$MONGO_LIST_INPROGRESS -e MONGO_LIST_DONE=$MONGO_LIST_DONE todoapp:test test_mongo.py
```

Comment:
May have said elsewhere, but you can shortcut this where the old/new environment variable names match, by just passing it once: `docker run -e MONGO_DB_NAME -e MONGO_CONN ... `